[kvernnn] iP

kvernnn opened the pull request:
# **Bao**
> The Next Generation of Personal Assistants

Bao is here to serve you! 

Bao is,
- Free 🤑 
- ~~Easy~~ Super easy to use 

## Bao's Capabilities
- [x] Track tasks (to-dos, deadlines and events!)
- [x] Search up tasks on-going at a specific date
- [x] Mark tasks as completed

### How to Use Bao
Bao supports the following commands:
- `todo`: Creates and tracks a to-do task
- `deadline`: Creates a deadline task
- `event`: Creates an event task
- `list`: Displays all tasks in the list
- `on`: Displays all tasks in the list occurring on the given date
- `mark`: Marks task at index as completed
- `unmark`: Unmarks task at index
- `delete`: Deletes task at index

## Download Bao
All you have to do to employ Bao is,
1. Download Bao [here](https://drive.google.com/file/d/1RcH4vkE2aUOl2KG6ioEPivgtgHRb5lTF/view?usp=sharing)
2. Unzip file
3. Run _java -jar ip.main.jar_ in the command line
4. Start using!

### Others
For fellow Java programmers, you can use it to practice Java too. Here's the main method:
```ruby
public static void main(String[] args) {
    new Bao(file_Path).run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kvernnn committed:
Renaming of Duke in README

kvernnn committed:
Edited greeting and exit of Bao

kvernnn committed:
Echo logic

kvernnn committed:
Added add and list logic

kvernnn committed:
Personalised Bao

kvernnn committed:
Use new class Task

kvernnn committed:
Mark and Unmark logic

kvernnn committed:
Implement ToDo as a child of Task class

kvernnn committed:
Minor renaming for readability

kvernnn committed:
Deadline feature

kvernnn committed:
Event feature

kvernnn committed:
TextUITesting

kvernnn committed:
Added exceptions (try and catch)

kvernnn committed:
Delete function using ArrayList

kvernnn committed:
Changing display of messages

kvernnn committed:
Implemented loadFile

kvernnn committed:
Added saveFile()

kvernnn committed:
Merge branch 'branch-Level-7' into master
Implemented level-7; added save and load functionalities

kvernnn committed:
Store deadline as LocalDateTime

kvernnn committed:
Bug fixes for LocalDateTime methods

kvernnn committed:
Search up tasks on a specific date

kvernnn committed:
Merge branch 'branch-Level-8' into master, added dates and times using LocalDateTime class

kvernnn committed:
Add Ui class to encapsulate messages to user

kvernnn committed:
Storage class to load and save files

kvernnn committed:
Revert "Storage class to load and save files"

This reverts commit 34b569b21bf09b04a994c3cd839301635305617d.

kvernnn committed:
Revert "Revert "Storage class to load and save files""

This reverts commit 2bff4ab84e2372f93c24fc463134fb6a1b5360c6.

kvernnn committed:
Add Parser and Storage classes

kvernnn committed:
Revert "Add Parser and Storage classes"

This reverts commit 6b2812fb3de9fe71acc672af9bd9e957e48ebbc1.

kvernnn committed:
Revert "Revert "Add Parser and Storage classes""

This reverts commit c5f70394013c703c45a8cd2c3d468cc4b8d30629.

kvernnn committed:
Edit Bao class to incorporate logic of Parser, Storage, TaskList and UI

kvernnn committed:
Merge branch 'branch-Level-9' into master

kvernnn committed:
Group classes into packages

kvernnn committed:
Format to follow JavaDoc

kvernnn committed:
Merge branch 'branch-A-Packages' into master

kvernnn committed:
Merge branch 'add-gradle-support' into master

kvernnn committed:
Create JUnit test cases for Bao and Task classes

kvernnn committed:
Merge branch 'branch-A-JUnit' into master

kvernnn committed:
Add comment and headers for Bao class

kvernnn committed:
Add comment and header for Parser class

kvernnn committed:
Add comment and header for Storage class

kvernnn committed:
Add comments and headers for TaskList class

kvernnn committed:
Add comment and header for UI class

kvernnn committed:
Add comment and header for Task class

kvernnn committed:
Add comment and header for ToDo class

kvernnn committed:
Add comment and header to Deadline class

kvernnn committed:
Add comment and header to Event class

kvernnn committed:
Fix indentation and coding standards

kvernnn committed:
Add Find functionality to Parser and TaskList class

kvernnn committed:
Merge branch 'branch-Level-9-v2' into master

kvernnn committed:
Deconflict branch 'branch-A-CodingStandard'

kvernnn committed:
Deconflict branch 'branch-A-JavaDoc'

kvernnn committed:
Format indentation

kvernnn committed:
Add checkstyle

kvernnn committed:
Merge branch 'branch-A-CheckStyle' into Master

kvernnn committed:
Debug for checkstyle

kvernnn committed:
Add GUI files

kvernnn committed:
Delete unnecessary Ui class

kvernnn committed:
Merge branch 'branch-Level-10' into Master

kvernnn committed:
Fix output indentation

reidenong commented on src/main/java/bao/main/Bao.java:
```diff
@@ -0,0 +1,127 @@
+
+    private void initialiseDates() {
+        inputDateFormat = new DateTimeFormatterBuilder()
+                .appendOptional(DateTimeFormatter.ofPattern("d/M/yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("dd/M/yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("d/MM/yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("d-MM-yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("dd-M-yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("d-M-yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("dd/MM/yyyy"))
+                .appendOptional(DateTimeFormatter.ofPattern("d/M/yyyy"))
+                .appendOptional(DateTimeFormatter.ofPattern("dd/M/yyyy"))
+                .appendOptional(DateTimeFormatter.ofPattern("d/MM/yyyy"))
+                .appendOptional(DateTimeFormatter.ofPattern("dd-MM-yyyy"))
+                .appendOptional(DateTimeFormatter.ofPattern("d-M-yyyy"))
+                .appendOptional(DateTimeFormatter.ofPattern("d-MM-yyyy"))
+                .appendOptional(DateTimeFormatter.ofPattern("dd-M-yyyy"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy-MM-d HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy-M-dd HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy-M-d HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy/MM/dd HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy/MM/d HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy/M/dd HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy/M/d HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy-MM-dd"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy-MM-d"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy-M-dd"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy-M-d"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy/MM/dd"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy/MM/d"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy/M/dc"))
+                .appendOptional(DateTimeFormatter.ofPattern("yyyy/M/d"))
+                .toFormatter();
+        fileDateFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+        dateOnlyFormat = DateTimeFormatter.ofPattern("MMM d yyyy");
+        outputDateFormat = DateTimeFormatter.ofPattern("MMM d yyyy, h:mm a");
+    }
```
```suggestion
    private void initialiseDates() {
        // Define the formats we want to support
        String[] patterns = {
            "d/M/yyyy HHmm", "dd/MM/yyyy HHmm", "dd/M/yyyy HHmm", "d/MM/yyyy HHmm",
            "dd-MM-yyyy HHmm", "d-MM-yyyy HHmm", "dd-M-yyyy HHmm", "d-M-yyyy HHmm",
            "dd/MM/yyyy", "d/M/yyyy", "dd/M/yyyy", "d/MM/yyyy",
            "dd-MM-yyyy", "d-M-yyyy", "d-MM-yyyy", "dd-M-yyyy",
            "yyyy-MM-dd HHmm", "yyyy-MM-d HHmm", "yyyy-M-dd HHmm", "yyyy-M-d HHmm",
            "yyyy/MM/dd HHmm", "yyyy/MM/d HHmm", "yyyy/M/dd HHmm", "yyyy/M/d HHmm",
            "yyyy-MM-dd", "yyyy-MM-d", "yyyy-M-dd", "yyyy-M-d",
            "yyyy/MM/dd", "yyyy/MM/d", "yyyy/M/d", "yyyy/M/d"
        };

        // Initialize the list of DateTimeFormatters
        inputDateFormats = new ArrayList<>();
        for (String pattern : patterns) {
            inputDateFormats.add(DateTimeFormatter.ofPattern(pattern));
        }

        fileDateFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
        dateOnlyFormat = DateTimeFormatter.ofPattern("MMM d yyyy");
        outputDateFormat = DateTimeFormatter.ofPattern("MMM d yyyy, h:mm a");
    }
```

This is an approach which may look cleaner?

reidenong commented on src/main/java/bao/task/Task.java:
```diff
@@ -0,0 +1,127 @@
+ * It is an abstract class that provides methods applicable for all types of tasks.
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
```
Leaving variables as ```protected``` should be fine in this use case, but they could be private as well given that we have getters and setters (eg. mark(), unmark())

reidenong reviewed (commented):
LGTM!

code looks nice, all changes suggested are mostly minor.

hyxnnii commented on src/main/java/bao/task/Task.java:
```diff
@@ -0,0 +1,127 @@
+     * @return Task object created from the string.
+     * @throws IllegalArgumentException Ff the string format is invalid or unknown task type is provided.
+     */
+    public static Task fromString(String string) {
```
perhaps this can be in a verb form eg) getTaskFromString?

hyxnnii commented on src/test/java/bao/task/TaskTest.java:
```diff
@@ -0,0 +1,137 @@
+     * Tests if todo task is created correctly using fromString method.
+     */
+    @Test
+    public void createToDoTaskUsingFromStringTest() {
```
perhaps for testing methods, instead of camel case, this convention can be used --unitBeingTested_descriptionOfTestInputs_expectedOutcome ?

hyxnnii reviewed (commented):
overall I think you did a great job :-)

kvernnn committed:
Build Bao.jar

kvernnn committed:
Use assert functions

kvernnn committed:
Improve code quality in Parser and Task classes

kvernnn committed:
Merge pull request #1 from kvernnn/branch-A-Assertions

Merge branch-A-Assertions into master branch

kvernnn committed:
Merge branch 'master' of https://github.com/kvernnn/ip

kvernnn committed:
Merge branch 'master' into branch-A-CodeQuality

kvernnn committed:
Merge pull request #2 from kvernnn/branch-A-CodeQuality

Merge branch-A-CodeQuality into master

kvernnn committed:
Merge branch 'master' of https://github.com/kvernnn/ip

kvernnn committed:
Clean up assert functions

Cleaned up unnecessary assert functions used in class (Parser, Main and TaskList).

This reduces clutter in the code, ensuring the assert functions are only used when necessary and allowing easier reading of the code.

kvernnn committed:
Add C-Tagging logic

kvernnn committed:
Merge branch 'branch-C-Tagging' into Master

kvernnn committed:
Remove user picture

kvernnn committed:
Set nicer margins, distinct colours and remove useless picture

kvernnn committed:
Merge branch 'branch-A-BetterGui' into Master

kvernnn committed:
Add name to bot

kvernnn committed:
Update README and screenshot

kvernnn committed:
Update UG and fix checkstyle errors

kvernnn committed:
Merge branch 'A-UserGuide-v2' into master
